helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementService.java

```diff
@@ -472,9 +472,15 @@ private MaintenanceManagementInstanceInfo takeFreeSingleInstanceHelper(String cl
   private List<String> batchHelixInstanceStoppableCheck(String clusterId,
       Collection<String> instances, Map<String, StoppableCheck> finalStoppableChecks,
       Set<String> toBeStoppedInstances) {
+
+    // Perform all but min_active replicas check in parallel
     Map<String, Future<StoppableCheck>> helixInstanceChecks = instances.stream().collect(
         Collectors.toMap(Function.identity(), instance -> POOL.submit(
             () -> performHelixOwnInstanceCheck(clusterId, instance, toBeStoppedInstances))));
+
+    // Perform min_active replicas check sequentially
+    addInstanceMinActiveReplicaCheck(helixInstanceChecks, toBeStoppedInstances);
```

Comment:
1. An easier approach that eliminates the need to check futureStoppableCheckByInstance is to perform the MIN_ACTIVE_REPLICA_CHECK_FAILED first. This way, there is no need to pass helixInstanceChecks as a parameter. 
2. On the other hand, be sure you check if the MIN_ACTIVE_REPLICA_CHECK_FAILED check is in the `_skipStoppableHealthCheckList`. If it is, the MIN_ACTIVE_REPLICA_CHECK_FAILED check should not be performed.

Reply:
`perform the MIN_ACTIVE_REPLICA_CHECK_FAILED first`
I think we need to perform the min active replica check last as this check is dependent on the stoppable status of other nodes. Specifically the result of checks instance_not_stable, has_error_partition, and has_disabled_partition will determine whether a sibling node will be stopped or not. If a node fails these checks, then it should be considered as not being stoppable when we perform the min_active check on one of its siblings

2. Good catch, thank you. I didn't account for this. Updated my PR to have a guard clause and return early if the min_active_replica check is in the `_skipStoppableHealthCheckList`

Reply:
Sounds good, let's keep the current order. What do you think about adding this check to the filterInstancesForNextCheck() method? Otherwise, we would end up with two identical methods, which seems redundant.

Reply:
Spoke offline with @MarkGaox , will look into leveraging filterInstancesForNextCheck and getInstanceHealthStatus in an attempt to reduce complexity added to code